File: App/app/src/main/java/hr/kcosic/app/model/entities/Location.kt
package hr.kcosic.app.model.entities

import hr.kcosic.app.model.bases.BaseEntity
import kotlinx.serialization.Serializable
import kotlin.properties.Delegates

@Serializable
class Location : BaseEntity() {
    var UserId: Int? = null
    var Longitude: Double? = null
    var Latitude: Double? = null
    var Street: String? = null
    var StreetNumber: String? = null
    var City: String? = null
    var County: String? = null
    var Country: String? = null

    @Serializable
    var Shops: MutableList<Shop>? = null

    @Serializable
    var User: User? = null

    override fun toString(): String {
        val street = if (Street.isNullOrEmpty()) "" else "$Street"
        val streetNumber = if (StreetNumber.isNullOrEmpty()) "" else " $StreetNumber"
        val city = if (City.isNullOrEmpty()) "" else ", $City"
        val country = if (Country.isNullOrEmpty()) "" else ", $Country"

        return "$street$streetNumber$city$country"
    }
}

File: WebAPI/WebAPI/Models/Partials/Location.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using WebAPI.Models.DTOs;

namespace WebAPI.Models.ORM
{

    partial class Location {
        public static LocationDTO ToDTO(Location item, int level)
        {
            if (item == null || level <= 0)
            {
                return null;
            }

            return new LocationDTO
            {
                DateCreated = item.DateCreated,
                DateDeleted = item.DateDeleted,
                DateModified = item.DateModified,
                Deleted = item.Deleted,
                Id = item.Id,
                Shops = Shop.ToListDTO(item.Shops, level - 1),
                User = User.ToDTO(item.User, level - 1),
                UserId = item.UserId,
                City = item.City,
                Country = item.Country,
                County = item.County,
                Latitude = item.Latitude,
                Longitude = item.Longitude,
                Street = item.Street,
                StreetNumber = item.StreetNumber,

            };
        }

        public static ICollection<LocationDTO> ToListDTO(ICollection<Location> list, int level)
        {
            if (list == null || level <= 0)
            {
                return null;
            }

            return list.Select(x => x.ToDTO(level)).ToList();
        }

        public LocationDTO ToDTO(int level)
        {
            return level > 0 ? new LocationDTO
            {

                DateCreated = DateCreated,
                DateDeleted = DateDeleted,
                DateModified = DateModified,
                Deleted = Deleted,
                Id = Id,
                Shops = Shop.ToListDTO(Shops, level),
                User = User.ToDTO(User, level),
                UserId = UserId,
                City = City,
                Country= Country,
                County= County,
                Latitude= Latitude,
                Longitude= Longitude,
                Street= Street,
                StreetNumber= StreetNumber,

            } : null;
        }
    }

}